clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4605,6 +5369,28 @@ void SemaSYCL::MarkDevices() {
   }
 }
 
+void SemaSYCL::ProcessFreeFunction(FunctionDecl *FD) {
+  if (isFreeFunction(*this, FD)) {
+    SyclKernelFieldChecker FieldChecker(*this);
```

Comment:
I am not sure if I mentioned this already but please look into if you can separate out the diagnostic checkers to where `CheckSYCLKernelCall` is called. This can be done in a follow-up PR if you are under a time crunch.

Reply:
The function CheckSYCLKernelCall is not called for free function invocation. So the diagnostic checker has to be where it is now.

Reply:
My point was you may need to move the diagnostic checkers to `CheckSYCLKernelCall` and call it for free functions or define a similar function in `SemaChecking` for free functions and call that.

Reply:
I'll make a note to move the disgnostics to a better position. Diagnostic checking is minimal right now. A future PR will be dedicated to diagnostic checking. Your suggestion can be incorporated into that PR.